grub-core/loader/i386/txt/txt.c: Use MAXPHYADDR in MTRR masks

miczyg1 opened the pull request:
Based on Intel TXT MLE Developer Guide revision 017.4 Table 4 the SINIT capabilities bit 8 indicates whether fixed 36bit masks or MAXPHYADDR masks are to be used in MTRR calculations. Failing to adhere to it may lead to creation of potentially disjoint WB cache ranges and violation of CRAM protections - according to the document.

miczyg1 committed:
grub-core/loader/i386/txt/txt.c: Use MAXPHYADDR in MTRR masks

Based on Intel TXT MLE Developer Guide revision 017.4 Table 4
the SINIT capabilities bit 8 indicates whether fixed 36bit masks
or MAXPHYADDR masks are to be used in MTRR calculations. Failing
to adhere to it may lead to creation of potentially disjoint WB
cache ranges and violation of CRAM protections - according to the
document.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

krystian-hebel commented:
This shouldn't make a difference with current Xen implementation, in https://github.com/TrenchBoot/xen/blob/aem-4.17.4/xen/arch/x86/boot/head.S#L131 36-bit is selected, and according to Table 5. Truth Table of SINIT / MLE functionality of TXT SDG (Revision 017.4) it takes precedence. To make it work, we should also set this bit in Xen, but currently [there are too many branches](https://github.com/TrenchBoot/trenchboot-issues/issues/48) that would require this change.

miczyg1 commented:
> This shouldn't make a difference with current Xen implementation, in https://github.com/TrenchBoot/xen/blob/aem-4.17.4/xen/arch/x86/boot/head.S#L131 36-bit is selected, and according to Table 5. Truth Table of SINIT / MLE functionality of TXT SDG (Revision 017.4) it takes precedence. To make it work, we should also set this bit in Xen, but currently [there are too many branches](https://github.com/TrenchBoot/trenchboot-issues/issues/48) that would require this change.

Yes, but then you end up with mixed MTRRs masks on a core. Some of them have MAXPHYADDR masks and some of them have 36bit masks... That's what I observed on VP6670

miczyg1 committed:
grub-core/loader/i386/txt: Set proper capabilities for CBnT

CBnT requires the auth PCR usage and no legacy PCR usage as per
TXT MLE Software Development Guide revision 017.4.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
